### DISCUSSION TITLE
Flet Authentication providers problem

### DISCUSSION THREAD

**Discussion by skarpir** [flet-dev/flet#2950] [General]
I'm having trouble with Authentication in Flet. I have tried 3 out of 4 the built in providers (Auth0, Github and Google). I have tried all example on https://flet.dev/docs/guides/python/authentication/. 

Everything seems to work fine but the page.on_login function is never called after I log in succesfully!

Has anyone else encountered this problem and found a solution?

**Comment by jersonjunior** [comment]
I'm having the same problem, I need to use Auth0 with keycloack, it works but I can't get user.id and other information from the logged-in user.

To run page.on_login=on_login tried like this page.on_login=on_login(), after the change error will occur due to (e), if you remove the (e) the function below does not work:

```
 def on_login(e):
        if e.error:
raise Exception(e.error)
        print("User ID:", page.auth.user.id)
        print("Access token:", page.auth.token.access_token) 
```

**Reply by jersonjunior** [reply]
Look this: 


https://github.com/flet-dev/flet/issues/2853


In Flet 0.21.0, because of migration to FastAPI web server, OAuth handler URL endpoint changed from **/api/oauth/redirect** to **/oauth_callback**. Use FLET_OAUTH_CALLBACK_HANDLER_ENDPOINT env var to customize.

**Reply by skarpir** [reply]
I finally had a chances to look at this and it works. 

Thank you for @jersonjunior

**Reply by SKD8** [reply]
> I finally had a chances to look at this and it works.
> 
> Thank you for @jersonjunior

Can you please provide the solution. i am having the same problem it works fine until i provide my consent via Google Oauth handler screen.  it just does not returns anything. it say page.auth.user is None.
